### PR TITLE
feature(core) add custom serializerInterface

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description
Add dedicated Serialize interface
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/core/src/Infrastructure/Serializer/SerializerInterface.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Core\Infrastructure\Serializer;
```
Comment (wfajczyk) [inline module/core/src/Infrastructure/Serializer/SerializerInterface.php:9]: I think it should been in  `SharedKernel` module

File: module/core/src/Infrastructure/Exception/DeserializationException.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Core\Infrastructure\Exception;
+
+class DeserializationException extends SerializerException
```
Comment (wfajczyk) [inline module/core/src/Infrastructure/Exception/DeserializationException.php:11]: I think it should been in `SharedKernel` module.

File: module/core/src/Infrastructure/Exception/SerializationException.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Core\Infrastructure\Exception;
+
+class SerializationException extends SerializerException
```
Comment (wfajczyk) [inline module/core/src/Infrastructure/Exception/SerializationException.php:11]: I think it should been in `SharedKernel` module.

**Review by wfajczyk — CHANGES_REQUESTED** [review]

File: module/core/src/Infrastructure/Serializer/SerializerInterface.php (hunk 1 above)
Comment (rprzedzik) [reply module/core/src/Infrastructure/Serializer/SerializerInterface.php]: I think this is not shared functionality other application can use own serializer matching needed requirements

File: module/core/src/Application/Serializer/SerializerInterface.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+namespace Ergonode\Core\Application\Serializer;
+
+interface SerializerInterface
+{
+    public const FORMAT = 'json';
+
+    /**
+     * @param mixed $object
+     */
+    public function serialize($object, ?string $format = self::FORMAT): string;
```
Comment (piotrkreft) [inline module/core/src/Application/Serializer/SerializerInterface.php:18]:
```suggestion
    public function serialize($data, ?string $format = self::FORMAT): string;
```

File: module/core/src/Application/Serializer/SerializerInterface.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    /**
+     * @return mixed
+     */
+    public function deserialize(string $json, string $type, ?string $format = self::FORMAT);
```
Comment (piotrkreft) [inline module/core/src/Application/Serializer/SerializerInterface.php:23]:
```suggestion
    public function deserialize(string $data, string $type, ?string $format = self::FORMAT);
```

**Review by piotrkreft — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Pull request merged**